feat(aio): search result styling, footer and refactor

sjtrimble opened the pull request:
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)



**What is the new behavior?**



**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...


**Other information**:

sjtrimble committed:
feat(aio): terminal code styling and code mixin

sjtrimble committed:
feat(aio): search result stying, add footer

sjtrimble committed:
feat(aio): add a and ol to standard typo styles

mary-poppins commented:
The angular.io preview for 27b2069 is available [here][1].

[1]: https://pr15411-27b2069493dfaa7b6050e30cd18ebf3cadf063f0.ngbuilds.io/

sjtrimble committed:
feat(aio): marketing responsive items and footer

sjtrimble committed:
feat(aio): footer styling

mary-poppins commented:
The angular.io preview for acadb46 is available [here][1].

[1]: https://pr15411-acadb464439efadbc93efe1b0a446c8247b1af24.ngbuilds.io/

petebacondarwin commented on aio/src/app/app.component.html:
```diff
@@ -8,13 +8,21 @@
+        <p>Powered by Google ©2010-2017. Code licensed under an <a href="/license">MIT-style License</a>.</p>
+        <p>Documentation licensed under <a href="http://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a>.
+        </p>
+        <p class="version-info">Version Info | {{ (versionInfo | async)?.full }}</p>
```
@sjtrimble did you agree this with @IgorMinar - Last I heard he was quite happy to have the version front and centre until we are closer to production.

petebacondarwin commented on aio/src/app/app.component.html:
```diff
@@ -8,7 +8,7 @@
 
   <md-sidenav #sidenav class="sidenav" [opened]="isSideBySide" [mode] = "isSideBySide ? 'side' : 'over'">
-    <aio-top-menu *ngIf="!isSideBySide" class="small" [nodes]="(navigationViews | async)?.TopBar" [homeImageUrl]="homeImageUrl"></aio-top-menu>
+    <aio-top-menu *ngIf="!isSideBySide" class="small" [nodes]="(navigationViews | async)?.TopBar" [homeImageUrl]="homeImageUrl"></aio-top-menu> 
```
trailing whitespace 🙀

petebacondarwin commented on aio/src/styles/2-modules/_code.scss:
```diff
@@ -1,13 +1,14 @@
 
-.code-example pre {
+.code-shell code {
+    @include codeblock($darkgray);
```
I think the padding and margins are a little off for this:

![screen shot 2017-03-23 at 10 48 29](https://cloud.githubusercontent.com/assets/15655/24244388/866395a4-0fb6-11e7-86f5-8ae02db8fdf2.png)

wardbell replied:
Why are we showing ANY headers in ALL CAPS.  Have your suggestions on h1-h5 not landed yet?

petebacondarwin commented on aio/src/styles/1-layouts/_search-results.scss:
```diff
@@ -1,15 +1,31 @@
+        }
+        &:visited {
+        text-decoration: none;
+        }
```
The search looks a lot nicer. We still need to do a bit of work with the padding maybe?

![screen shot 2017-03-23 at 10 51 30](https://cloud.githubusercontent.com/assets/15655/24244436/c831be5c-0fb6-11e7-9e5f-9e8111553248.png)

Also if the search is longer than the page then it is not possible to scroll down the results.

petebacondarwin commented on aio/src/styles/1-layouts/_footer.scss:
```diff
@@ -1,8 +1,8 @@
     z-index: 10;
-    background-color: $blue;
+    background-image: url('../src//assets/images/backgrounds/ng-pattern-sm.png');
+    background-repeat: repeat;
```
Loving the new footer :-)

wardbell replied:
Me too. I think it needs some whitespace above it
![image](https://cloud.githubusercontent.com/assets/129061/24260283/a8a82cc0-0fb0-11e7-90fd-10a94ae49162.png)

petebacondarwin commented on aio/src/app/app.component.html:
```diff
@@ -8,13 +8,21 @@
 <md-sidenav-container class="sidenav-container">
 
   <md-sidenav #sidenav class="sidenav" [opened]="isSideBySide" [mode] = "isSideBySide ? 'side' : 'over'">
-    <div class="version-info vertical-menu-item">{{ (versionInfo | async)?.full }}</div>
```
perhaps we should leave this in place until we get close to production?

wardbell replied:
@petebacondarwin Stef moved it to the footer (ln 25) and Igor is cool with that.

wardbell replied:
I'm seeing a blue artifact like this on the mini toolbar.

Also, IMO, we should have a tinier logo ... or, better, none at all here ... and bigger font for the text.
![mini-nav-toolbar](https://cloud.githubusercontent.com/assets/129061/24259529/5bd23352-0fae-11e7-9c0b-01a0f54e7b82.png)

petebacondarwin reviewed (changes requested):
This looks very nice!
I made some minor comments that can be improved in later PRs.
I think we need to check about the version info thing with @IgorMinar before we change it.

wardbell commented on aio/src/styles/mixins.scss:
```diff
@@ -6,4 +6,12 @@
   transition-timing-function: ease-in-out;
+}
+
+@mixin codeblock($bgcolor) {
```
Remind me to factor this into our revised syntax highlighting PR

wardbell reviewed (commented):
My feedback. Not withholding approval because my suggestions could come in a follow up PR. Want to keep moving forward !

mary-poppins commented:
The angular.io preview for 1618a91 is available [here][1].

[1]: https://pr15411-1618a917c7868b3fd316789222f4aeba5d93de2e.ngbuilds.io/

sjtrimble commented:
1. I just fixed the last commit message to follow the guidelines.

2. You don't want headers in caps? That was part of my style summary we all reviewed a while back. I can change it if you think it's not appropriate. I have noticed that often h3s are used when h2s could be used, which does change the look and feel of things.

3. @wardbell I don't see the blue outline on the mini toolbar, but I do see something similar. I can look into it some more (I was going to ask you if you knew what that was).

4. I will push in another commit shortly for some of the other smaller items you mentioned.

5. @petebacondarwin I did check about the version number with Igor, which is why I moved it.

@wardbell This is the goal for the side nav menu when collapsed. @petebacondarwin said he would help with the logic on this one. Only change would be the the logo should be just the badge icon, not the full logo. This would address one of your comments.

<img width="326" alt="screen shot 2017-03-22 at 11 50 02 pm" src="https://cloud.githubusercontent.com/assets/2958442/24261426/2b781806-0fb4-11e7-939c-08fb43297bed.png">

sjtrimble committed:
feat(aio): search results and code margins styling

mary-poppins commented:
The angular.io preview for 425634f is available [here][1].

[1]: https://pr15411-425634ff5f2146b3b50a0942e14ccb0feafa83c8.ngbuilds.io/

sjtrimble committed:
feat(aio): search input animation

sjtrimble committed:
feat(aio): block code examples style fix

mary-poppins commented:
The angular.io preview for d9a06b8 is available [here][1].

[1]: https://pr15411-d9a06b884fffa194270ce3314abef7d8f0f4bba6.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 28db8fd is available [here][1].

[1]: https://pr15411-28db8fdfa5fc22f40d7753d6ac88da75c0485154.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 9eb9cad is available [here][1].

[1]: https://pr15411-9eb9cad09311a340264dddc98ecb8625815ee929.ngbuilds.io/
